File: Project/NKJMovieComposerDemo/Classes/ConfirmViewController.m
//
//  ConfirmViewController.m
//
//  Created by nakajijapan.
//  Copyright 2014 nakajijapan. All rights reserved.
//

#import "ConfirmViewController.h"

@interface ConfirmViewController ()
@property MPMoviePlayerController *movielayer;
@property AppDelegate *appDelegate;
@end

@implementation ConfirmViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    [self.view setBackgroundColor:[UIColor whiteColor]];
    
    self.appDelegate = [[UIApplication sharedApplication]delegate];
    NSLog(@"composedMoviePath = %@", self.appDelegate.composedMoviePath);
    NSURL* movieUrl = [NSURL fileURLWithPath:self.appDelegate.composedMoviePath];
    
    _movielayer = [[MPMoviePlayerController alloc] initWithContentURL:movieUrl];
    [_movielayer setControlStyle:MPMovieControlStyleEmbedded];
    [_movielayer setScalingMode:MPMovieScalingModeAspectFit];
    [_movielayer setShouldAutoplay:NO];
    [_movielayer.view setBackgroundColor:[UIColor lightGrayColor]];
    [_movielayer.view setFrame:CGRectMake(0.0f, 74.0f, 320, 320)];
    [_movielayer prepareToPlay];
    
    [self.view addSubview:_movielayer.view];
}

@end

File: Project/NKJMovieComposerDemo/Classes/ConfirmViewController.swift
//
//  ConfirmViewController.swift
//  NKJMovieComposerDemo
//
//  Created by nakajijapan on 2014/06/11.
//  Copyright (c) 2014 net.nakajijapan. All rights reserved.
//

import UIKit
import MediaPlayer

class ConfirmViewController: UIViewController {
    
    var appDelegate:AppDelegate!
    var movielayer: MPMoviePlayerController!
    
    override init(nibName nibNameOrNil: String!, bundle nibBundleOrNil: NSBundle!) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
    }
    
    required init(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.view.backgroundColor = UIColor.whiteColor()
        
        self.appDelegate = UIApplication.sharedApplication().delegate as AppDelegate
        println("composedMoviePath = \(self.appDelegate.composedMoviePath)")
        let movieUrl = NSURL(fileURLWithPath: self.appDelegate.composedMoviePath)

        movielayer = MPMoviePlayerController(contentURL: movieUrl)
        movielayer.controlStyle = MPMovieControlStyle.Embedded
        movielayer.scalingMode = MPMovieScalingMode.AspectFill
        movielayer.view.backgroundColor = UIColor.lightGrayColor()
        movielayer.view.frame = CGRect(x: 0, y: 74, width: 320, height: 320)
        movielayer.prepareToPlay()
        
        self.view.addSubview(movielayer.view)
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
}
